src/pal/pal_open_enclave.h

```diff
@@ -114,19 +55,16 @@ namespace snmalloc
       oe_abort();
     }
 
-    template<bool committed>
-    static void* reserve(size_t size, size_t align) noexcept
+    static std::pair<void*, size_t>
+    reserve_at_least(size_t request_size) noexcept
     {
-      // The following are all true from the current way snmalloc uses the PAL.
-      // The implementation here is depending on them.
-      SNMALLOC_ASSERT(size == bits::next_pow2(size));
-      SNMALLOC_ASSERT(align == bits::next_pow2(align));
-      if (size != align)
-        error("Critical assumption violated!");
-
       FlagLock lock(spin_lock);

```

Comment:
I had to read this a couple of times, so it probably needs a comment.  If I read it correctly, this is just giving the entire heap to the `AddressSpaceManager` to handle.  That's very neat and should simplify the cross-process case.

Reply:
Yep. It should also bring down the memory usage on snmalloc, as we get as much aligned memory as possible, and put the meta-data in the bits below superslab alignment.